tests/test_restart.py

```diff
@@ -0,0 +1,445 @@
+#!/usr/bin/env python
+''' VIC exact restart testing '''
+
+import numpy as np
+import datetime
+import os
+import glob
+import xarray as xr
+from test_utils import read_vic_ascii
+
+
+# -------------------------------------------------------------------- #
+def prepare_restart_run_periods(restart_dict, state_basedir, statesec):
+    ''' For restart tests, read full running period and splitting dates into
+    datetime objects
+
+    Parameters
+    ----------
+    restart_dict: <class 'configobj.Section'>
+        A section of the config file for exact restart test setup
+    state_basedir: <str>
+        Basedirectory of output state files.
+        For restart tests, state files will be output as:
+        <state_basedir>/<run_start_date>_<run_end_date>/<state_file>
+    statesec: <int>
+        STATESEC option in global parameter file
+
+    Returns
+    ----------
+    run_periods: OrderedDict
+        A list of running periods, including the full-period run, and all
+        splitted runs in order. Each element of run_period is a dictionary with
+        keys:
+            start_date
+            end_date
+            init_state  # None, or full path of the initial state file
+                        # e.g., '/path/19490101_19490105/states_19490105_82800'
+
+    '''
+
+    # --- Read in full running period --- #
+    start_date = datetime.datetime(restart_dict['start_date'][0],
+                                   restart_dict['start_date'][1],
+                                   restart_dict['start_date'][2])
+    end_date = datetime.datetime(restart_dict['end_date'][0],
+                                 restart_dict['end_date'][1],
+                                 restart_dict['end_date'][2])
+    # --- Identify each of the splitted running period --- #
+    list_split_dates = []
+    for i, split in enumerate(restart_dict['split_dates']):
+        list_split_dates.append(
+            datetime.datetime(restart_dict['split_dates'][split][0],
+                              restart_dict['split_dates'][split][1],
+                              restart_dict['split_dates'][split][2]))
```

Comment:
my comments on how you structure the configuration file will help simplify this section.